jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1205,10 +1205,15 @@ public final Module getUnnamedModule()
  */
 final Class<?> loadClass(Module module, String className) {
 	Class<?> localClass = null;
-	try {
-		localClass = loadClassHelper(className, false, false, module);
-	} catch (ClassNotFoundException e) {
-		// returns null if the class can't be found
+	
+	if ((bootstrapClassLoader == null) || (this == bootstrapClassLoader)) {
```

Comment:
Shouldn't the fix go into loadClassHelper() so other callers of loadClassHelper() get the same benefits?

Reply:
Do you mean moving this code snippet from `loadClass (className)` and `loadClass(module, className)` into `loadClassHelper()`?
`BOOT_LOADER` extends `BuiltinClassLoader` which overrides `loadClass(className, resolveClass)`, hence `BOOT_LOADER.loadClass (className)` will invoked this overridden method and missed the checking for bootstrap classloader within `loadClassHelper()` hence can't call `VMAccess.findClassOrNull(className, bootstrapClassLoader)` for bootstrap classes. I think this was the main reason we put this checking code snippet within `loadClass (className)`. This PR adds the check for `loadClass(module, className)` as well.
This is for bootstrap classloader class loading, other classloader use scenario doesn't need this check.